scripts/move_patterns.py

```diff
@@ -0,0 +1,268 @@
+  list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+* Neither the name of display_node nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+
+"""
+Example client program for sending multiple move_basic commands
+"""
+
+import rospy
+
+# our custom messages for the commands we will be using
+import getopt, sys
+python3 = True if sys.hexversion > 0x03000000 else False
+import genpy
+import struct
+
+import geometry_msgs.msg
+import actionlib
+from   move_base_msgs.msg import MoveBaseAction, MoveBaseGoal
+import tf
+
+import traceback
+import time
+
+def printUsage():
+    print "-h --help       - This help menu"
+    print "-c --continue   - Continuous operation, no prompts to continue"
+    print "-w --waypoints  - Select a waypoint list by name as below"
+    print "                  line         A simple line"
+    print "                  box          A square box"
+    print "                  octagon      An octagon pattern"
+    print "                  figure8      A figure 8 dual octagon"
+    print "-s --scale      - A different scale for the pattern "
+    print "-x --offsetX    - An offset for X map placement of the pattern" 
+    print "-y --offsetY    - An offset for Y map placement of the pattern" 
+
+
+class Controller:
+
+
+    #######################################################################################
+    # Define lists of waypoints in terms of X,Y values in meters and robot angular yaw pose
+    # The entries on each waypoint are as follows:
+    #   1) X coordinate in the map in meters
+    #   2) Y coordinate in the map in meters
+    #   3) Angle in radians that the robot assumes once at X,Y
+    #   4) A simple text comment
+
+    # A simple line of one meter length
+    figureLine = [\
+       [ 0.00,  0.00,  0.000, "MOVE: Leg A" ], \
+       [ 0.50,  0.00,  0.000, "MOVE: Leg B" ] \
+       ]
```

Comment:
You don't need the `\` in Python.  Also tuples `(` ... `)` are better than lists `[` ... `]` since they are immutable

Reply:
This is python 2.7.   I'll give it a try but I thought I had run into grief in the past on python end of lines.